src/averageAges.js

```diff
@@ -15,6 +15,26 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
+  if (!century) {
+    const arrOfMen = [...people].filter((person) => person.sex === 'm');
+
+    const averageAges = arrOfMen.map((person) => person.died - person.born);
+
+    return averageAges.reduce((accumulator, age) =>
+      (accumulator + age
+      ), 0) / averageAges.length;
+  }
+
+  const arrOfMenWithCentury = [...people].filter((person) =>
+    person.sex === 'm' && Math.ceil(person.died / 100)
+    === century);
+
+  const averageAgesWithCentury = arrOfMenWithCentury.map((person) =>
+    person.died - person.born);
+
+  return averageAgesWithCentury.reduce((accumulator, age) =>
+    (accumulator + age
+    ), 0) / averageAgesWithCentury.length;
```

Comment:
follow DRY, you repeat EVERYTHING second time if there is century, better combine checks in filters and remove your if